src/sparsify/auto/tasks/finetune/finetune.py

```diff
@@ -0,0 +1,320 @@
+
+        """
+        if os.path.exists(dataset_path):
+            if Path(dataset_path).suffix not in [".yaml", ".yml"]:
+                raise RuntimeError(
+                    "LLMFinetuner expects a yaml file compliant with llmfoundry."
+                )
+            with open(dataset_path) as yaml_file:
+                self._train_config = om.load(yaml_file)
+        else:
+            raise FileNotFoundError(
+                f"{dataset_path} does not exist. Plase ensure "
+                " the yaml file exists and the path provided is correct."
+            )
+
+        if self._train_config.get("loggers"):
+            for _, log_config in self._train_config["loggers"].items():
+                if "log_dir" in log_config:
+                    log_config["log_dir"] = os.path.join(log_dir, log_config["log_dir"])
+                else:
+                    log_config["log_dir"] = log_dir
+
+        self._train_config.save_folder = os.path.join(
+            train_directory, Path(self._train_config.save_folder)
+        )
+        self._model_name = self._train_config["model"]["name"]
+        self._validate_yaml()
+
+    @property
+    def model_name(self) -> str:
+        """
+        :return: model name for the LLM
+        """
+        return self._model_name
+
+    def _validate_yaml(self):
+        """
+        Validate that the provided yaml is compatible with llmfoundry.
+        """
+        if not self._train_config.get("train_loader"):
+            raise ValueError(
+                "the provided config file is missing details on the train_loader"
+            )
+
+        data_loaders = [self._train_config.get("train_loader")]
+        if self._train_config.get("eval_loader"):
+            data_loaders.append(self._train_config.get("eval_loader"))
+
+        for loader in data_loaders:
+            if loader["name"] == Datatypes.TEXT.value:
+                if self.model_name in TEXT_DENOISING_MODELS:
+                    raise ValueError(
+                        f"Model type {self.model_name} is not supported "
+                        " for text dataloaders. Please use the "
+                        " text_denoising dataloader."
+                    )
+            elif loader["name"] == Datatypes.TEXT_DENOISING.value:
+                if self.model_name in TEXT_MODELS:
+                    raise ValueError(
+                        f"Model type {self.model_name} is not supported "
+                        " for text_denoising dataloaders. Please use the "
+                        " text dataloader."
+                    )
+
+    def _build_model(self, tokenizer: PreTrainedTokenizerBase) -> HuggingFaceModel:
+        """
+        Based on the model name, pull and return the pretrained hugging face model.
+
+        :param tokenizer: transformers tokenizer
+        :return: HuggingFaceModel from the mosaicml composer library
+        """
+        if not COMPOSER_MODEL_REGISTRY.get(self.model_name):
```

Comment:
nit: does this read better?
```suggestion
        if self.model_name not in COMPOSER_MODEL_REGISTRY :
```